### ISSUE TITLE
[client] Accessibility: Announce the content I'm annotating

### ISSUE THREAD

**Issue by chrisshaw** [hypothesis/product-backlog#1344]
**As a user using assistive technologies like NVDA, JAWS, or VoiceOver, I want to be notified of the content related to the annotation I'm editing, so that I  can move on confidently to engage in discussions more quickly.**

This showed up as a major defect in a recent accessbility review:

> Could hear what was being typed in the editor, but no way of knowing what was selected unless the screen reader user went back and re-read the whole page.

Our WCAG auditor mention recommended we label the highlighted content so users know what they are annotating or editing:

> Add an aria-describedby attribute to the text box that point to the highlighted text so that assistive technology users know what information they are annotating or replying to. This would be especially useful when users of assistive technologies such as screen readers create a new annotation.

## Related

Related to #1341

**Comment by chrisshaw** [comment]
**Issue was linked to Harvestr Discovery:**  [Know what annotation I'm editing](https://app.harvestr.io/components/0/list/ddEYBCOeN)

**Comment by lyzadanger** [comment]
Am I understanding this correctly: A user has highlighted some text  in the "host page" and either clicked "Annotate" or typed "A" to create an annotation. They are taken to the edit interface in the sidebar and the annotation-content textarea input is focused. But they are unable to tell what the selected/highlighted text is that they are annotating?

The suggested approach of `aria-labelledby` might not work here (if I'm grokking correctly) as the selected text is in a different document. So this one will take a little more investigation to come up with an appropriate solution. We do have access to the selected text in the sidebar where the edit interface is, so we likely have other ways of setting up ARIA "glue" to what the selected text, if indeed I am understanding the problem correctly. cc @chrisshaw

**Comment by lyzadanger** [comment]
Note: Dependency is inaccurate.

**Comment by chrisshaw** [comment]
@lyzadanger I updated all the dependency statements to related as you're right.

And yes, you're grokking it correctly. That's the complaint as described.

**Comment by lyzadanger** [comment]
### “Before”

Before we set out on any recent accessibility enhancements in the sidebar, Voiceover on Mac OS would read out the following after pressing “A” to annotate on some selected text (once the user has been moved to the associated `textarea`:

![449D2B5A-D9E9-4C1A-9359-7C6B06B70EB3](https://user-images.githubusercontent.com/439947/176184980-d93cf55b-27af-423a-90bf-fd96b257546d.png)

If a user wanted more context, they could navigate outward. They’d first hit the markdown editor toolbar, then the quoted annotation text:

![D48B7872-46D1-4189-BE5C-CF9F8AAB1705](https://user-images.githubusercontent.com/439947/176185037-b440daf7-6613-4b5f-88e1-73fe30c6a5e4.png)

Heading outward/upward a couple more slots would get them to the annotation’s container `article` element:

![CF887DE8-8B58-4001-B80A-3182E118A07F](https://user-images.githubusercontent.com/439947/176185067-6a8aa33b-9169-4fe7-9390-9d57b29970a3.png)

### “After”

After recent work, the pathway is a little more clear.

After typing “A” to annotate, there is more description:

![7C957907-5ABB-487E-87E1-CEF9C229B8DF](https://user-images.githubusercontent.com/439947/176185166-71270897-35b1-4939-b6e6-e1227aa65cc6.png)

Navigating up/out still moves through the toolbar, then the quoted text:

![D48B7872-46D1-4189-BE5C-CF9F8AAB1705](https://user-images.githubusercontent.com/439947/176185205-989dcd26-7986-4ccd-bdc5-e2596304a1ef.png)

The containing annotation `article` is better labeled:

![2739D31A-FF7E-44B5-8AF4-B44F98A9C75E](https://user-images.githubusercontent.com/439947/176185223-bc0a59e0-dedf-4ef8-a2e0-a3f7c2094e77.png)

Thus, the screen-reader structure (for lack of a better term) of an annotation can be represented something like this:

```
<article aria-label="New annotation by Yours Truly">
  <h3><link to user /></h3>
  <link to group />
  <blockquote>"I don't admit that a fresh illustration is an explanation," said I with some asperity.</blockquote>
  <markdown toolbar / >
  <textarea aria-label="Annotation body" / >
  (tag editor, publish controls, footer...)
</article>
```

This is roughly the same view a sighted user has, conceptually. This hierarchy seems roughly correct to me.

When put into the annotation-body `textarea` for a new annotation, if a user (one not using assistive technology) loses track of the text they’ve just selected, they’ll “glance up” above the editor and see the quoted text. Likewise, I think a user with a screen-reader would analogously try navigating up a slot or two to get context.

For sighted users, the annotation is visually bounded in a “card”. That same “bounding” is represented by the (now descriptively-labeled) semantic `article` element for those using screen readers.

I believe this is an appropriate amount of context, based on what I can find in real-world examples and ARIA documentation, _on the sidebar side of things_.

I believe that there is room for improvement on the _annotator_ side of the equation, WRT discoverability of keyboard commands, e.g. — but that falls outside of the scope of this request, which is concerned with access to context for a new annotation. I believe that context is provided. Happy to revisit if an auditor or helpful user can give us precise direction. I do not believe that `aria-describedby` (the suggested attribute to use for this request) is applicable in this case.
